obspy/signal/spectral_estimation.py

```diff
@@ -1285,6 +1297,191 @@ def _add_npz(self, filename):
             msg = msg % (duplicates, len(_times_processed), filename)
             warnings.warn(msg)
 
+    def plot_spectrogram(self, cmap=obspy_sequential, clim=None, grid=True,
+                         filename=None, show=True):
+        """
+        Plot the temporal evolution of the PSD in a spectrogram-like plot.
+
+        .. note::
+            For example plots see the :ref:`Obspy Gallery <gallery>`.
+
+        :type cmap: :class:`matplotlib.colors.Colormap`
+        :param cmap: Specify a custom colormap instance. If not specified, then
+            the default ObsPy sequential colormap is used.
+        :type clim: list
+        :param clim: Minimum/maximum dB values for lower/upper end of colormap.
+            Specified as type ``float`` or ``None`` for no clipping on one end
+            of the scale (e.g. ``clim=[-150, None]`` for a lower limit of
+            ``-150`` dB and no clipping on upper end).
+        :type grid: bool
+        :param grid: Enable/disable grid in histogram plot.
+        :type filename: str
+        :param filename: Name of output file
+        :type show: bool
+        :param show: Enable/disable immediately showing the plot.
+        """
+        import matplotlib.pyplot as plt
+        from matplotlib.dates import date2num
+
+        xedges = date2num(
+            [t.datetime for t in self.times_processed] +
+            [(self.times_processed[-1] + self.ppsd_length).datetime])
+        yedges = self.period_xedges
+        meshgrid_x, meshgrid_y = np.meshgrid(xedges, yedges)
+        data = np.array(self.psd_values).T
+
+        fig = plt.figure()
+        ax = fig.add_subplot(1, 1, 1)
+
+        quadmesh = ax.pcolormesh(meshgrid_x, meshgrid_y, data, cmap=cmap,
```

Comment:
with this technique, how will gaps be plotted ? interpolation between previous-next psd available ?